CMSgov/easi-app
# [EASI-2515] TRB Advice Letter Submitted Email

# EASI-2515

## Changes and Description

- Added backend logic, automated test, and resolver logic to send an email when a TRB advice letter is submitted

## How to test this change

Follow these steps to set up an advice letter and submit it, then navigate to localhost:1080 to verify the email in MailCatcher
![image](https://user-images.githubusercontent.com/95702411/216469663-62729841-8f9c-4c46-a5dc-0238d4c782ea.png)

```GraphQL
# Make a TRB request and get it to a proper state
mutation {
  createTRBRequest(requestType: NEED_HELP) {
    id
    type
  }
}

# Add a consult meeting
mutation {
  updateTRBRequestConsultMeetingTime(
    input: {
      trbRequestId: "trbIdFromAbove"
      consultMeetingTime: "2021-01-01T19:15:15.187687Z"
      notes: "bla bla"
      copyTrbMailbox: true
      notifyEuaIds: []
    }
  ) {
    consultMeetingTime
    taskStatuses {
      adviceLetterStatus
    }
  }
}

# Add a lead
mutation {
  updateTRBRequestTRBLead(
    input: { trbLead: "ABCD", trbRequestId: "trbIdFromAbove" }
  ) {
    id
    trbLead
  }
}

# Create an advice letter, take note of the ID
mutation {
  createTRBAdviceLetter(trbRequestId: "trbIdFromAbove") {
    id
    trbRequestId
    createdBy
    createdAt
  }
}

mutation {
  updateTRBAdviceLetter(
    input: {
      trbRequestId: "trbIdFromAbove"
      meetingSummary: "some stuff was talked about"
      nextSteps: "do things"
      isFollowupRecommended: true
      followupPoint: "about six months from now"
    }
  ) {
    id
    trbRequestId
    meetingSummary
    nextSteps
    isFollowupRecommended
    followupPoint
    dateSent
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}

# Submit the advice letter, this is where the email should be sent
mutation {
  sendTRBAdviceLetter(
    input: {
      id: "adviceLetterIdFromAbove"
      copyTrbMailbox: true
      notifyEuaIds: []
    }
  ) {
    id
    trbRequestId
    meetingSummary
    nextSteps
    isFollowupRecommended
    followupPoint
    dateSent
    createdBy
    createdAt
    modifiedBy
    modifiedAt
  }
}

```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
